omnigibson/prims/rigid_prim.py

```diff
@@ -306,7 +307,29 @@ def get_angular_velocity(self):
         """
         return self._rigid_prim_view.get_angular_velocities()[0]
 
-    def set_position_orientation(self, position=None, orientation=None):
+    def set_position_orientation(
+        self, position=None, orientation=None, frame: Literal["world", "parent", "scene"] = "world"
+    ):
+        """
+        Set the position and orientation of XForm Prim.
+
+        Args:
+            position (None or 3-array): The position to set the object to. If None, the position is not changed.
+            orientation (None or 4-array): The orientation to set the object to. If None, the orientation is not changed.
+            frame (Literal): The frame in which to set the position and orientation. Defaults to world. parent frame
+            set position relative to the object parent. scene frame set position relative to the scene.
+        """
+
+        # If we are in a scene, compute the scene-local transform before setting the pose
+        if frame == "scene" and self.scene is not None:
+
+            # if position or orientation is None, get the current position and orientation relative to scene
+            current_position, current_orientation = self.get_position_orientation(frame="scene")
+            position = current_position if position is None else np.array(position, dtype=float)
+            orientation = current_orientation if orientation is None else np.array(orientation, dtype=float)
+
+            position, orientation = T.pose_transform(*self.scene.prim.get_position_orientation(), position, orientation)
```

Comment:
do we have to repeat these lines of code?

Reply:
Does this look like a good way of implementing this helper function? If I place this helper function into tranform_helper.py, the set_position_orientation API wil need to pass in "self" into the helper function. 
```
def compute_scene_local_transform(prim, frame, position, orientation):
    if frame == "scene" and prim.scene is not None and position is None and orientation is None:
        position, orientation = prim.get_position_orientation(frame="scene")
        position = position if position is None else np.array(position, dtype=float)
        orientation = orientation if orientation is None else np.array(orientation, dtype=float)
        position, orientation = T.pose_transform(*prim.scene.prim.get_position_orientation(), position, orientation)
    return position, orientation
```